apps/wallet/src/ui/app/hooks/useDeepBook.ts

```diff
@@ -145,28 +158,246 @@ export function useBalanceConversion(
 	balance: BigInt | BigNumber | null,
 	base: Coins,
 	quote: Coins,
+	conversionRate: number = 1,
 ) {
 	const { data: averagePrice, ...rest } = useAveragePrice(base, quote);
+	const averagePriceWithConversion = averagePrice.shiftedBy(conversionRate);
 
 	const rawValue = useMemo(() => {
-		if (!averagePrice || !balance) return null;
+		if (!averagePriceWithConversion || !balance) return null;
 
-		const rawUsdValue = new BigNumber(balance.toString()).multipliedBy(averagePrice).toNumber();
+		const rawUsdValue = new BigNumber(balance.toString())
+			.multipliedBy(averagePriceWithConversion)
+			.toNumber();
 
 		if (isNaN(rawUsdValue)) {
 			return null;
 		}
 
 		return rawUsdValue;
-	}, [averagePrice, balance]);
+	}, [averagePriceWithConversion, balance]);
 
 	return {
 		rawValue,
-		averagePrice,
+		averagePrice: averagePriceWithConversion,
 		...rest,
 	};
 }
 
-export function useSuiBalanceInUSDC(suiBalance: BigInt | BigNumber | null) {
-	return useBalanceConversion(suiBalance, Coins.SUI, Coins.USDC);
+export function useCreateAccount({
+	onSuccess,
+	deepBookClient,
+}: {
+	onSuccess: () => void;
+	deepBookClient: DeepBookClient;
+}) {
+	const activeAccount = useActiveAccount();
+	const activeAccountAddress = activeAccount?.address;
+	const signer = useSigner(activeAccount);
+
+	return useMutation({
+		mutationKey: [DEEPBOOK_KEY, 'create-account', activeAccountAddress],
+		mutationFn: async () => {
+			if (activeAccountAddress) {
+				const txb = deepBookClient.createAccount(activeAccountAddress);
+
+				return signer?.signAndExecuteTransactionBlock({ transactionBlock: txb });
+			}
+
+			return null;
+		},
+		onSuccess,
+	});
+}
+
+export function useMarketAccountCap(activeAccountAddress?: string) {
+	const {
+		data,
+		isLoading: getOwnedObjectsLoading,
+		isError: getOwnedObjectsError,
+		isRefetching: isRefetchingOwnedObjects,
+		refetch,
+	} = useGetOwnedObjects(
+		activeAccountAddress,
+		{
+			MatchAll: [{ StructType: '0xdee9::custodian_v2::AccountCap' }],
+		},
+		10,
```

Comment:
Why do you pass a limit of 10 then only use the first result?

Reply:
fixed